theme.json

```diff
@@ -0,0 +1 @@
+lib/theme.json
```

Comment:
I'd actually move the file instead of adding a symlink

Reply:
This is explained in #66027

WP-CLI assumes that `theme.json` only exists in a theme, and in themes it's required to be in the root. It does not look for any other subfolders and I don't wanna change that in WP-CLI.

The Gutenberg plugin is an exception because it holds the "source of truth" for `theme.json` and does so in the `lib` folder.

Reply:
I need to catch up a bit on this, but last I checked ([3 years ago](https://meta.trac.wordpress.org/ticket/5779#comment:11)) it was working fine (it extracted strings from core, Gutenberg, and themes).

Reply:
Yes please test it again. But the way WP-CLI works hasn't changed: https://github.com/wp-cli/i18n-command/blob/0133eb1326e7e0613315dd94c347a2a03ee6f2d4/src/ThemeJsonExtractor.php#L19-L25

Reply:
> WP-CLI assumes that theme.json only exists in a theme, and in themes it's required to be in the root. It does not look for any other subfolders and I don't wanna change that in WP-CLI.

🤔  This is not how it worked @swissspidy At least not what I [did](https://github.com/wp-cli/i18n-command/pull/254/files#diff-730d4162be1a82ede4e934e60764e07f104a11c609ef985c354bf0c586c2388b). Though I see there's now [different logic](https://github.com/wp-cli/i18n-command/blob/main/src/ThemeJsonExtractor.php#L19). In addition to themes, there are strings in WordPress ([theme.json](https://github.com/WordPress/wordpress-develop/blob/trunk/src/wp-includes/theme.json)) and Gutenberg ([theme.json](https://github.com/WordPress/gutenberg/blob/trunk/lib/theme.json)).

I've checked GlotPress, and I see strings coming up from `theme.json` for themes (checked TwentyTwentyFive). Both the source (`theme.json` file) and the context (the badge below the string) are picked are correctly:

https://github.com/user-attachments/assets/bba11c9b-966b-4a15-8029-dab1e5ea061f

However, if I do the same search for WordPress core and Gutenberg, some other strings pop up:

| Gutenberg | WordPress |
| --- | --- |
| <img width="1879" alt="Screenshot 2024-12-18 at 17 32 29" src="https://github.com/user-attachments/assets/2589f145-2576-4e87-aa46-6a7a33fbe7bd" /> | <img width="1879" alt="Screenshot 2024-12-18 at 17 32 13" src="https://github.com/user-attachments/assets/59df7885-b013-4e91-9703-b0a30e62c91c" /> |

Reply:
It's a bit difficult to track commits to specific Pull Request in the i18n-command file, so sorry if I'm misunderstanding this: judging by [this commit](https://github.com/wp-cli/i18n-command/commit/345eb7a88443d4db63a8d8e7c7aac935fb197392) it sounds we haven't been extracting strings from the `theme.json` provided by WordPress (or Gutenberg) since 2 years ago?

Reply:
I've also searched for specific strings that are only defined in theme.json. Example: "Dark grayscale" for duotone. Same result, they are not present in translate:

| Gutenberg | WordPress |
| --- | --- |
| <img width="1879" alt="Screenshot 2024-12-18 at 17 43 23" src="https://github.com/user-attachments/assets/5406285a-d599-425b-828e-d1b8fec07fb2" /> | <img width="1879" alt="Screenshot 2024-12-18 at 17 43 02" src="https://github.com/user-attachments/assets/8a4e29bb-1b8e-46b0-b7b9-dc8ade6903d1" /> |

Reply:
There're additional things going on. Despite [WordPress core](https://translate.wordpress.org/projects/wp/dev/) and the [TwentyTwentyFour](https://translate.wordpress.org/projects/wp-themes/twentytwentyfour/) theme being 100% translated to Galician, note how the duotone strings provided by theme (and [translated](https://translate.wordpress.org/projects/wp-themes/twentytwentyfour/gl/default/?filters%5Bterm%5D=black+and+sandstone&filters%5Bterm_scope%5D=scope_any&filters%5Bstatus%5D=current_or_waiting_or_fuzzy_or_untranslated&filters%5Buser_login%5D=&filter=Apply+Filters&sort%5Bby%5D=priority&sort%5Bhow%5D=desc)) are not used. It is still displayed "Black and sandstone":

<img width="716" alt="Screenshot 2024-12-18 at 17 59 25" src="https://github.com/user-attachments/assets/d1bcd647-ddfe-49c2-853b-9a6e1851f7f6" />

Reply:
More context: for testing, avoid using color gradients, palette, or font sizes. Some of these are also defined in the client ([see](https://github.com/WordPress/gutenberg/blob/trunk/packages/block-editor/src/store/defaults.js#L43)) so it can give you false positives.

Reply:
> However, if I do the same search for WordPress core and Gutenberg, there are no strings coming from `theme.json`

With that current logic in WP-CLI that would be expected. As I said, it only looks for a specific location at the moment.

That change was made with the assumption that `theme.json` is supposed to live only in themes and in the root folder of theme. This helped avoid an issue where, when extracting strings in core, `theme.json` strings from all the default themes were erroneously included (because they live in `wp-content`.

I suppose we didn't realize at the time that `wp-includes/theme.json` also exists... 🤔 

> it sounds we haven't been extracting strings from the theme.json provided by WordPress (or Gutenberg) since 2 years ago?

Yes, you're right. Hmm we need to make some changes to WP-CLI then and to Meta ([`sites/trunk/wordpress.org/public_html/wp-content/plugins/wporg-gp-customizations/inc/cli/class-make-core-pot.php`](https://meta.trac.wordpress.org/browser/sites/trunk/wordpress.org/public_html/wp-content/plugins/wporg-gp-customizations/inc/cli/class-make-core-pot.php))

Reply:
Closing the loop here: we're continuing the conversation at https://github.com/wp-cli/i18n-command/issues/423